contrib/epee/include/net/net_utils_base.h

```diff
@@ -89,6 +89,48 @@ namespace net_utils
 	inline bool operator>=(const ipv4_network_address& lhs, const ipv4_network_address& rhs) noexcept
 	{ return !lhs.less(rhs); }
 
+	class ipv6_network_address
+	{
+		std::string m_ip;
+		uint16_t m_port;
+
+	public:
+		ipv6_network_address(const std::string& ip, uint16_t port) noexcept
+			: m_ip(ip), m_port(port) {}
+
+		bool equal(const ipv6_network_address& other) const noexcept;
+		bool less(const ipv6_network_address& other) const noexcept;
+		constexpr bool is_same_host(const ipv6_network_address& other) const noexcept
+		{ return ip() == other.ip(); }
```

Comment:
This trips -Winvalid-constexpr. To me it looks like it's because they have different lifetimes, which is not permitted.